flink-ml-python/pyflink/ml/lib/feature/tests/test_binarizer.py

```diff
@@ -0,0 +1,81 @@
+#  or more contributor license agreements.  See the NOTICE file
+#  distributed with this work for additional information
+#  regarding copyright ownership.  The ASF licenses this file
+#  to you under the Apache License, Version 2.0 (the
+#  "License"); you may not use this file except in compliance
+#  with the License.  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+# limitations under the License.
+################################################################################
+import os
+
+from pyflink.common import Types
+
+from pyflink.ml.core.linalg import Vectors, DenseVectorTypeInfo
+from pyflink.ml.lib.feature.binarizer import Binarizer
+from pyflink.ml.tests.test_utils import PyFlinkMLTestCase
+
+
+class BinarizerTest(PyFlinkMLTestCase):
+    def setUp(self):
+        super(BinarizerTest, self).setUp()
+        self.input_data_table = self.t_env.from_data_stream(
+            self.env.from_collection([
+                (0,
+                 Vectors.dense(2.1, 3.1)),
+                (2,
+                 Vectors.dense(3.3, 1.3)),
+            ],
+                type_info=Types.ROW_NAMED(
+                    ['f0', 'f1'],
+                    [Types.INT(), DenseVectorTypeInfo()])))
+
+        self.expected_output_data_1 = Vectors.dense(0.0, 1.0)
+        self.expected_output_data_2 = Vectors.dense(1.0, 0.0)
+
+    def test_param(self):
+        binarizer = Binarizer()
+
+        binarizer.set_input_cols('f0', 'f1') \
+            .set_output_cols('of0', 'of1') \
+            .set_thresholds(1.5, 2.5)
+
+        self.assertEqual(('f0', 'f1'), binarizer.input_cols)
+        self.assertEqual(('of0', 'of1'), binarizer.output_cols)
+        self.assertEqual((1.5, 2.5), binarizer.get_thresholds())
+
+    def test_save_load_transform(self):
+        binarizer = Binarizer() \
+            .set_input_cols('f0', 'f1') \
+            .set_output_cols('of0', 'of1') \
+            .set_thresholds(1.5, 2.5)
+
+        path = os.path.join(self.temp_dir, 'test_save_load_transform_binarizer')
+        binarizer.save(path)
+        binarizer = Binarizer.load(self.t_env, path)
+
+        output_table = binarizer.transform(self.input_data_table)[0]
+        actual_outputs = [(result[0], result[2], result[3]) for result in
+                          self.t_env.to_data_stream(output_table).execute_and_collect()]
+
+        self.assertEqual(2, len(actual_outputs))
+        for actual_output in actual_outputs:
+            self.assertEqual(2, len(actual_output[2]))
+            if actual_output[0] == 0:
+                self.assertAlmostEqual(0.0, actual_output[1])
```

Comment:
It would be better to explicitly set a tolerance threshold to the `assertAlmostEqual`.